src/main/java/de/tum/in/test/api/ast/type/ExceptionHandlingType.java

```diff
@@ -0,0 +1,41 @@
+package de.tum.in.test.api.ast.type;
+
+import java.util.Map;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.stmt.*;
+
+public enum ExceptionHandlingType implements Type {
+	/**
+	 * Returns a list with all exception handling types
+	 */
+	ANY(Map.of("Assert-Statement", AssertStmt.class, "Throw-Statement", ThrowStmt.class, "Try-Statement", TryStmt.class,
+			"Catch-Statement", CatchClause.class)),
+	/**
+	 * Returns to assert type
+	 */
+	ASSERT(Map.of("Assert-Statement", AssertStmt.class)),
+	/**
+	 * Returns the throw type
+	 */
+	THROW(Map.of("Throw-Statement", ThrowStmt.class)),
+	/**
+	 * Returns the try-catch exception handling type
+	 */
+	TRYCATCH(Map.of("Try-Statement", TryStmt.class, "Catch-Statement", CatchClause.class));
```

Comment:
I am still not sure if including `try` here makes sense. I would suggest including only `catch` because `catch` is the only parts of try-catch-final which caches the flow of exceptions. Try with resources will become a more common pattern for scoped resources, and that may have little to do with exceptions.

Reply:
[Implemented part of the new feedback](https://github.com/ls1intum/Ares/pull/248/commits/ead5d75f0814dbd3738c57b058de8759b980e46d): Removed `TryStmt.class` and renamed enum element name to `CATCH`.